politeiawww/user.go

```diff
@@ -1582,46 +1582,110 @@ func (p *politeiawww) processManageUser(mu *v1.ManageUser, adminUser *user.User)
 }
 
 // processUsers returns a list of users given a set of filters.
-func (p *politeiawww) processUsers(users *v1.Users) (*v1.UsersReply, error) {
+func (p *politeiawww) processUsers(users *v1.Users, isAdmin bool) (*v1.UsersReply, error) {
 	var reply v1.UsersReply
-	reply.Users = make([]v1.AbridgedUser, 0)
+	reply.Users = make([]v1.AbridgedUser, 0, v1.UserListPageSize)
 
 	emailQuery := strings.ToLower(users.Email)
 	usernameQuery := formatUsername(users.Username)
+	pubkeyQuery := users.PublicKey
 
-	err := p.db.AllUsers(func(user *user.User) {
-		reply.TotalUsers++
-		userMatches := true
+	// Ensure we got a proper pubkey, if provided
+	_, err := validatePubkey(pubkeyQuery)
+	if pubkeyQuery != "" && err != nil {
+		return nil, err
+	}
+
+	// Get user ID by pubkey, if provided
+	var userID string
+	u, err := p.db.UserGetByPublicKey(pubkeyQuery)
+	if pubkeyQuery != "" && err != nil {
+		fmt.Printf("%v", "user not found")
+		return nil, err
+	} else if pubkeyQuery != "" && err == nil {
+		userID = u.ID.String()
+		fmt.Printf("%v", userID)
+	}
+
+	if isAdmin {
+
```

Comment:
Good link. Putting in notes for inclusion in docs at some point. There's long been talk of an intro page specifically for devs.

Reply:
Another extremely common mistake that people make when they're coming from a dynamically typed language like javascript is making names too verbose.

It might be worth including this link as well.

https://talks.golang.org/2014/names.slide#1